### PR TITLE
[develop] Fix AWS "filename too long"

### LINKED ISSUE #4279: FileSystemException: File name too long
When running a sufficiently nested workflow, I get the error `File name too long`, because WDL is generating a temporary file named `temp-s3-6826696772619511254cromwell-execution_best_practise_3bf9c436-31f7-4d86-ba87-cf54248e05bc_call-ConvertPairedFastQsToUnmappedBamWf_unmap.ConvertPairedFastQsToUnmappedBamWf_e10704e8-7ff4-4e4c-89ad-48270734ba25_call-CreateFoFN_write_lines_238fd975f9e3d166dbec07b20ad88c51.tmp`, which is greater than 256 characters and thus too large for any Linux system.

I'm running Cromwell 36 on with the AWS Batch backend. The traceback is:

```
2018-10-19 01:29:11,310 cromwell-system-akka.dispatchers.backend-dispatcher-8681 ERROR - Failed command instantiation
java.lang.Exception: Failed command instantiation
        at cromwell.backend.standard.StandardAsyncExecutionActor.instantiatedCommand(StandardAsyncExecutionActor.scala:565)
        at cromwell.backend.standard.StandardAsyncExecutionActor.instantiatedCommand$(StandardAsyncExecutionActor.scala:500)
        at cromwell.backend.impl.aws.AwsBatchAsyncBackendJobExecutionActor.instantiatedCommand$lzycompute(AwsBatchAsyncBackendJobExecutionActor.scala:74)
        at cromwell.backend.impl.aws.AwsBatchAsyncBackendJobExecutionActor.instantiatedCommand(AwsBatchAsyncBackendJobExecutionActor.scala:74)  
        at cromwell.backend.standard.StandardAsyncExecutionActor.commandScriptContents(StandardAsyncExecutionActor.scala:313)
        at cromwell.backend.standard.StandardAsyncExecutionActor.commandScriptContents$(StandardAsyncExecutionActor.scala:312)
        at cromwell.backend.impl.aws.AwsBatchAsyncBackendJobExecutionActor.commandScriptContents(AwsBatchAsyncBackendJobExecutionActor.scala:74)
        at cromwell.backend.impl.aws.AwsBatchAsyncBackendJobExecutionActor.batchJob$lzycompute(AwsBatchAsyncBackendJobExecutionActor.scala:132) 
        at cromwell.backend.impl.aws.AwsBatchAsyncBackendJobExecutionActor.batchJob(AwsBatchAsyncBackendJobExecutionActor.scala:131)
        at cromwell.backend.impl.aws.AwsBatchAsyncBackendJobExecutionActor.executeAsync(AwsBatchAsyncBackendJobExecutionActor.scala:342)
        at cromwell.backend.standard.StandardAsyncExecutionActor.executeOrRecover(StandardAsyncExecutionActor.scala:943)
        at cromwell.backend.standard.StandardAsyncExecutionActor.executeOrRecover$(StandardAsyncExecutionActor.scala:935)
        at cromwell.backend.impl.aws.AwsBatchAsyncBackendJobExecutionActor.executeOrRecover(AwsBatchAsyncBackendJobExecutionActor.scala:74)
        at cromwell.backend.async.AsyncBackendJobExecutionActor.$anonfun$robustExecuteOrRecover$1(AsyncBackendJobExecutionActor.scala:65)
        at cromwell.core.retry.Retry$.withRetry(Retry.scala:38)
        at cromwell.backend.async.AsyncBackendJobExecutionActor.withRetry(AsyncBackendJobExecutionActor.scala:61)
        at cromwell.backend.async.AsyncBackendJobExecutionActor.cromwell$backend$async$AsyncBackendJobExecutionActor$$robustExecuteOrRecover(AsyncBackendJobExecutionActor.scala:65)
        at cromwell.backend.async.AsyncBackendJobExecutionActor$$anonfun$receive$1.applyOrElse(AsyncBackendJobExecutionActor.scala:88)
        at scala.PartialFunction$OrElse.applyOrElse(PartialFunction.scala:172)
        at scala.PartialFunction$OrElse.applyOrElse(PartialFunction.scala:172)
        at akka.actor.Actor.aroundReceive(Actor.scala:517)
        at akka.actor.Actor.aroundReceive$(Actor.scala:515)
        at cromwell.backend.impl.aws.AwsBatchAsyncBackendJobExecutionActor.aroundReceive(AwsBatchAsyncBackendJobExecutionActor.scala:74)
        at akka.actor.ActorCell.receiveMessage(ActorCell.scala:588)
        at akka.actor.ActorCell.invoke(ActorCell.scala:557)
        at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:258)
        at akka.dispatch.Mailbox.run(Mailbox.scala:225)
        at akka.dispatch.Mailbox.exec(Mailbox.scala:235)
        at akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
        at akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
        at akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
        at akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
Caused by: common.exception.AggregatedMessageException: Error(s):
Could not evaluate expression: write_lines(array_of_files): [Attempted 1 time(s)] - FileSystemException: /tmp/temp-s3-6826696772619511254cromwell-execution_best_practise_3bf9c436-31f7-4d86-ba87-cf54248e05
bc_call-ConvertPairedFastQsToUnmappedBamWf_unmap.ConvertPairedFastQsToUnmappedBamWf_e10704e8-7ff4-4e4c-89ad-48270734ba25_call-CreateFoFN_write_lines_238fd975f9e3d166dbec07b20ad88c51.tmp: File name too lon
g
        at common.validation.Validation$ValidationTry$.toTry$extension1(Validation.scala:68)
        at common.validation.Validation$ValidationTry$.toTry$extension0(Validation.scala:64)
        at cromwell.backend.standard.StandardAsyncExecutionActor.instantiatedCommand(StandardAsyncExecutionActor.scala:563)
        ... 31 common frames omitted
```

It seems that each "segment" of this filename, e.g. `bc_call-ConvertPairedFastQsToUnmappedBamWf_unmap.ConvertPairedFastQsToUnmappedBamWf_e10704e8-7ff4-4e4c-89ad-48270734ba25`, should probably be in a different folder, not just as a different segment of the same filename.

### PR DESCRIPTION
**Pull request by aednichols**
Closes #4279

### CODE REVIEW

File: filesystems/s3/src/main/java/org/lerch/s3fs/S3Channel.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package org.lerch.s3fs;
```
Comment (geoffjentry) [inline filesystems/s3/src/main/java/org/lerch/s3fs/S3Channel.java]: If we're making changes here we'll also want to make a PR to [emil's repo](https://github.com/elerch/Amazon-S3-FileSystem-NIO2)

Comment (aednichols) [reply filesystems/s3/src/main/java/org/lerch/s3fs/S3Channel.java]: Thanks for the tip, will do

Comment (aednichols) [reply filesystems/s3/src/main/java/org/lerch/s3fs/S3Channel.java]: https://github.com/Upplication/Amazon-S3-FileSystem-NIO2/pull/125

Comment (aednichols) [reply filesystems/s3/src/main/java/org/lerch/s3fs/S3Channel.java]: https://github.com/elerch/Amazon-S3-FileSystem-NIO2/pull/2

**Comment by geoffjentry** [comment]
Merging this for @aednichols as he's out, this is green, and there are no outstanding things described to clean up.

**Comment by geoffjentry** [comment]
@aednichols tho i did not delete the branch just in case, will leave that to you :)

**Pull request merged**